lib/python/treadmill_aws/garbage_collector.py

```diff
@@ -0,0 +1,75 @@
+"""AWS specific treadmill garbage collector
+"""
+
+import logging
+from datetime import datetime
+from datetime import timedelta
+
+import jmespath
+
+from treadmill import admin
+from treadmill import context
+from treadmill_aws.awscontext import GLOBAL
+
+_LOGGER = logging.getLogger(__name__)
+_VALID_INSTANCES = None
+
+
+def _valid_ec2_instances():
+    global _VALID_INSTANCES  # pylint: disable=W0603
```

Comment:
Removed lru_cache for this function as suggested, but then I had to call global _VALID_INSTANCES to share the result of this function with other functions